csrc/scheduler/matmul.cpp

```diff
@@ -57,54 +57,73 @@ void moveInnerBroadcastLeft(TensorView* tv, int number_of_inner_pos = 3) {
   tv->reorder(order_map);
 }
 
+// Utility to check concrete static size:
+inline void check_concrete_static_dim(IterDomain* id) {
+  TORCH_INTERNAL_ASSERT(
+      !id->isBroadcast() && !id->isReduction(),
+      "no support for reduction or broadcast domains, but got ",
+      id->toString());
+  TORCH_INTERNAL_ASSERT(
+      id->extent()->isConstInt(),
+      "swizzled dimension's extend must be known during scheduling, got ",
+      id->toString());
+}
+
 //! Automatically generates the shared memory swizzled data layout
-//!  for matmul mainloop.
-//! The shared mem datalayout is always 2D currently, and this utility
-//!  function assumes that the innermost 2 dimensions on shared_mem_tv
-//!  are the ones begin swizzled.
-void prologSwizzle(TensorView* shared_mem_tv, const MatmulParams& params) {
+//!  for matmul mainloop and epilogue.
+//! The shared mem data layout is always 2D currently, and this utility
+//!  function assumes that the shared_mem_tv has the following structure:
+//!  [tile_row, tile_col, ***shift***] where the parameter `shift` is the number
```

Comment:
Are all the shifts reductions? If so, can we compute the value of shift within `swizzleSharedMemory`?

Reply:
yes. changed to compute the value of shift within `swizzleSharedMemory`.